src/orders/src/main/java/mushop/orders/services/OrdersService.java

```diff
@@ -93,6 +100,23 @@ public CustomerOrder createNewOrder(NewOrderResource orderPayload) {
             }
             if (!paymentResponse.isAuthorised()) {
                 meterRegistry.counter("orders.rejected","cause","payment_declined").increment();
+                // Send Payment failure metrics to OCI Monitoring..
+                try {
+                    System.out.println("In Order Service, Setting up OCI Monitoring .. ");
+                    monitoringClient = monitoringConfig.initConnection();
+                    if (monitoringClient == null) {
+                        System.out.printf("Cannot send metrics to OCI Monitoring, Ensure you have set the required ENV variables tenantId, userId, privateKey and fingerprint ..");
```

Comment:
monitoring client could be wrapped in an Option(MonitoringClient) and case matched ?
But this service is doing way more than order processing at this point.

Reply:
I'll discuss on the new service option suggested.